tests/src/main/java/testcases/KT006_reference_self/Child.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2023 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package testcases.KT006_reference_self
+
+@motif.Scope
+interface Child {
+
+    @motif.Objects
+    abstract class Objects : ObjectComponent<Bar>
```

Comment:
This is the key ingredient of the repro case, that an `Objects` inherits a function that contains a self-referencing generic parameter